target_chains/sui/contracts/sources/pyth_accumulator.move

```diff
@@ -0,0 +1,158 @@
+        let ema_price = deserialize::deserialize_i64(message_cur);
+        let ema_conf = deserialize::deserialize_u64(message_cur);
+        let price_info = price_info::new_price_info(
+            clock::timestamp_ms(clock) / 1000, // not used anywhere kept for backward compatibility
+            clock::timestamp_ms(clock) / 1000,
+            price_feed::new(
+                price_identifier,
+                pyth::price::new(price, conf, expo, publish_time),
+                pyth::price::new(ema_price, ema_conf, expo, publish_time),
+            )
+        );
+        price_info
+    }
+
+    // parse_and_verify_accumulator_updates takes as input a merkle root and cursor over the encoded update message (containing encoded
+    // leafs and merkle proofs), iterates over each leaf/proof pair and verifies it is part of the tree, and finally outputs the set of
+    // decoded and authenticated PriceInfos.
+    fun parse_and_verify_accumulator_updates(cursor: &mut Cursor<u8>, merkle_root: Bytes20, clock: &Clock): vector<PriceInfo> {
+        let update_size = deserialize::deserialize_u8(cursor);
+        let price_info_updates: vector<PriceInfo> = vector[];
+        while (update_size > 0) {
+            let message_size = deserialize::deserialize_u16(cursor);
+            let message = deserialize::deserialize_vector(cursor, (message_size as u64)); //should be safe to go from u16 to u16
+            let message_cur = cursor::new(message);
+            let price_info = parse_price_feed_message(&mut message_cur, clock);
+            cursor::destroy_empty(message_cur);
+            vector::push_back(&mut price_info_updates, price_info);
+
+            // isProofValid pops the next merkle proof from the front of cursor and checks if it proves that message is part of the
+            // merkle tree defined by merkle_root
+            assert!(merkle_tree::isProofValid(cursor, merkle_root, message), E_INVALID_PROOF);
+            update_size = update_size - 1;
+        };
+        price_info_updates
+    }
+
+    #[test_only]
+    public fun test_get_price_feed_updates_from_accumulator(accumulator_message: vector<u8>, wormhole_state: &WormState, clock: &Clock): vector<PriceInfo> {
+        let vaa_bytes = parse_vaa_bytes_from_accumulator_message(copy accumulator_message);
+
+        let cur = cursor::new(accumulator_message);
+        let header: u32 = deserialize::deserialize_u32(&mut cur);
+
+        if ((header as u64) != PYTHNET_ACCUMULATOR_UPDATE_MAGIC) {
+            abort 99
+        };
+
+        let verified_vaa = vaa::parse_and_verify(wormhole_state, vaa_bytes, clock);
+
+        // main entry-point
+        let price_infos = parse_and_verify_accumulator_message(&mut cur, vaa::take_payload(verified_vaa), clock);
+
+        //TODO -  update_cache(updates);
+        // cursor should be empty after parsing the header and accumulator message
+        cursor::destroy_empty(cur);
+        price_infos
+    }
```

Comment:
It is only used for one test? What is the purpose of it compared with using `get_verified_vaa_from_accumulator_message` and calling the contract methods?

Reply:
this was written earlier and used solely for testing `pyth_accumulator.move`, but I can remove it now